var/spack/repos/builtin/packages/rocm-openmp-extras/package.py

```diff
@@ -0,0 +1,337 @@
+        versions_dict.setdefault(item, {})[components[inner_index]] = \
+            component_hashes[inner_index][outer_index]
+
+
+class RocmOpenmpExtras(Package):
+    """OpenMP support for ROCm LLVM."""
+
+    homepage = tools_url + "/aomp"
+    url = tools_url + "/aomp/archive/rocm-uc-3.10.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala', 'estewart08']
+    version('3.10.0', sha256=versions_dict['3.10.0']['aomp'])
+    version('3.9.0', sha256=versions_dict['3.9.0']['aomp'])
+
+    depends_on('cmake@3:', type='build')
+    depends_on('mesa18~llvm@18.3:', type=('build', 'link'))
+    depends_on('py-setuptools', type='build')
+    depends_on('python@3:', type='build')
+    depends_on('perl-data-dumper', type='build')
+    depends_on('awk', type='build')
+    depends_on('elfutils', type=('build', 'link'))
+    depends_on('libffi', type=('build', 'link'))
+
+    for ver in ['3.9.0', '3.10.0']:
+        depends_on('hsakmt-roct@' + ver, type=('build', 'run'), when='@' + ver)
+        depends_on('comgr@' + ver, type='build', when='@' + ver)
+        depends_on('hsa-rocr-dev@' + ver, type=('build', 'run'),
+                   when='@' + ver)
+        depends_on('rocm-device-libs@' + ver, type=('build', 'run'),
+                   when='@' + ver)
+        depends_on('llvm-amdgpu@' + ver, type=('build', 'run'),
+                   when='@' + ver)
+
+        resource(
+            name='rocm-device-libs',
+            url=compute_url +
+            '/ROCm-Device-Libs/archive/rocm-uc-' + ver + '.tar.gz',
+            sha256=versions_dict[ver]['devlib'],
+            expand=True,
+            destination='rocm-openmp-extras',
+            placement='rocm-device-libs',
+            when='@' + ver)
+
+        resource(
+            name='flang',
+            url=tools_url + '/flang/archive/rocm-uc-' + ver + '.tar.gz',
+            sha256=versions_dict[ver]['flang'],
+            expand=True,
+            destination='rocm-openmp-extras',
+            placement='flang',
+            when='@' + ver)
+
+        resource(
+            name='aomp-extras',
+            url=tools_url + '/aomp-extras/archive/rocm-uc-' + ver + '.tar.gz',
+            sha256=versions_dict[ver]['extras'],
+            expand=True,
+            destination='rocm-openmp-extras',
+            placement='aomp-extras',
+            when='@' + ver)
+
+        resource(
+            name='llvm-project',
+            url=compute_url + '/llvm-project/archive/rocm-' + ver + '.tar.gz',
+            sha256=versions_dict[ver]['llvm'],
+            expand=True,
+            destination='rocm-openmp-extras',
+            placement='llvm-project',
+            when='@' + ver)
+
+    def setup_run_environment(self, env):
+        devlibs_prefix = self.spec['rocm-device-libs'].prefix
+        openmp_extras_prefix = self.spec['rocm-openmp-extras'].prefix
+        llvm_prefix = self.spec['llvm-amdgpu'].prefix
+        env.set('AOMP', '{0}'.format(format(llvm_prefix)))
```

Comment:
I don't think all of these double `format`s do anything, you only need a single `format`

Reply:
Indeed, this was overlooked.